src/spawn_stubs.c

```diff
@@ -488,6 +502,188 @@ CAMLprim value spawn_unix(value v_env,
     }
   }
 
+  if (new_mask || old_mask)
+    pthread_sigmask(SIG_SETMASK, new_mask, old_mask);
+}
+
+#if defined(USE_POSIX_SPAWN)
+
+CAMLprim value spawn_unix(value v_env,
```

Comment:
I don't know if it makes sense in this case, especially since code has already been written, but:
we generally try to put logic on the OCaml side if we can.
In `unix_spawn` we couldn't do that (while staying sane) since forking interacts poorly with OCaml runtime.
I'm thinking `posix_spawn` doesn't share that problem, so we could move much of this code to OCaml, instead creating C bindings for the individual `posix_spawnattr_*` functions.

Reply:
Interesting. Yeah, I see how that could be good. On the other hand, I could extend the blocking section around `posix_spawn` to include all the related `posix_spawn*` calls, if you think it's helpful to release the runtime system for longer.

Reply:
Extending the blocking section is also a liability because you can't access the OCaml heap (or raise exceptions) from the blocking section. For that reason I'm not exactly excited about extending it. (although if some of those functions are known to be blocking/slow then that may be worth it)